profiles.wordpress.org/public_html/wp-content/plugins/wporg-5ftf-activity-handler/wporg-5ftf-activity-handler.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/*
+Plugin Name: WordPress.org 5ftf Activity Handler
+Plugin URI: http://wordpress.org
+License: GPL2
+Version: 1.1
+Description: Handles saving of last 5ftf contribution.
+*/
+
+defined( 'ABSPATH' ) or die();
+
+if ( ! class_exists( 'WPOrg_5ftf_Activity_Handler' ) ) {
+
+	class WPOrg_5ftf_Activity_Handler {
+
+		/**
+		 *  Name of meta key that tracks last contribution as a unix timestamp.
+		 *
+		 * @var string
+		 */
+		const last_contribution_meta_key = 'wporg_5ftf_last_contribution';
+
+		/**
+		 * Constructor.
+		 */
+		public function __construct() {
+			add_filter( 'bp_activity_add', array( $this, 'handle_activity' ) );
+			add_filter( 'wporg_github_added_activity', array( $this, 'handle_github_activity' ) );
```

Comment:
```suggestion
			if ( 'profiles.wordpress.org' === site_url() ) {
				add_filter( 'bp_activity_add', array( $this, 'handle_activity' ) );
			} elseif ( str_ends_with( 'wordpress.org', site_url() ) {
				add_filter( 'wporg_github_added_activity', array( $this, 'handle_github_activity' ) );
			}
```

That would help self-document the underlying assumptions.

That `elseif` would probably cover all the sites in the w.org multi-network in practice, but it's not really future proof, since we could add something like `doaction.org` in the future. It also matches sites in other networks that end in the domain, which we might want to activate the plugin on in the future.

Since we can't use any of the constants in the shared directory, so we might want to add something like an `IS_WPORG_MULTINETWORK` constant to `wp-config.php`.

Reply:
Very helpful. I'm not sure I'll get this right the first time since I lack the context.

I think we agree on the approach, so I'll get this PR in a better state and ping you then. Thanks for the help so far 👍

Reply:
Could we not look for the presence of the filter?

`has_filter( 'wporg_github_added_activity', 'example_alter_the_content' )`

Reply:
Nevermind, I misunderstood how that works. WP Newb :) 🤕

Reply:
https://github.com/WordPress/wordpress.org/pull/78/commits/91f82f239f097eff37cd7886f5fd4cd6af258142

Reply:
Pretty close; I don't think w/ need the hostname check now that we have the constant, and it's a little safer to check the value of the constant:

```php
} elseif ( defined( 'IS_WPORG_MULTINETWORK' ) && IS_WPORG_MULTINETWORK ) {
```

Reply:
Ha so close 👌.